apps/user/services/authentication.go

```diff
@@ -48,12 +54,32 @@ func (service AuthenticationService) Authorize(c *gin.Context) (
 		return nil, errors.New("invalid state")
 	}
 
-	// TODO replace with oauth logic
-	email, _ := url.QueryUnescape(c.Query("code"))
-	user := userModels.User{}
+	tok, err := googleOauthConf.Exchange(context.TODO(), c.Query("code"))
+	if err != nil {
+		logrus.Info("Error occurred while exchanging code with token")
+		return nil, err
+	}
+
+	client := googleOauthConf.Client(context.TODO(), tok)
+
+	googlePlusAPIRequester, err := plus.New(client)
+	if err != nil {
+		return nil, err
+	}
 
-	if err := db.Where("email = ?", email).First(&user).Error; err != nil {
-		logrus.Info(fmt.Sprintf("User with email %s not found", email), err)
+	authenticatedGoogleUser, err := googlePlusAPIRequester.People.Get("me").Do()
```

Comment:
authenticatedGoogleUser could be googleUser

Reply:
updated to use `googleUser`